libos/src/bookkeep/libos_handle.c

```diff
@@ -325,8 +328,9 @@ struct libos_handle* detach_fd_handle(uint32_t fd, int* flags,
         handle = __detach_fd_handle(handle_map->map[fd], flags, handle_map);
 
     unlock(&handle_map->lock);
-
-    (void)clear_posix_locks(handle);
+    if (handle->id == 0 || handle->ref_count == 1) {
```

Comment:
I was enabling MongoDB in gramine and hit a segmentation fault issue with this line. It seems null pointer check of the handle is missing here. Could you please fix this?

Reply:
Thanks for your reply, I have fixed it.

Reply:
Thanks.

Reply:
Hi, Do you found some bugs about this PR so far?